base/MAPL_GridManager.F90

```diff
@@ -78,14 +79,83 @@ subroutine add_prototype(this, grid_type, prototype)
       
    end subroutine add_prototype
 
+   ! Is prototype_name present in the prototypes map keys?
```

Comment:
I presume that the current logic completely skipped this check?   Good catch if so.